pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -16,261 +16,266 @@ package extensionresources
-				return validation.ValidateWorker(n.(*extensionsv1alpha1.Worker))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateWorkerUpdate(n.(*extensionsv1alpha1.Worker), o.(*extensionsv1alpha1.Worker))
-			},
-		},
+// AddWebhooks add extension's valdation webhook to manager
+func AddWebhooks(mgr manager.Manager) error {
+	for obj, validator := range validators {
+		if err := builder.WebhookManagedBy(mgr).WithValidator(validator).For(obj).Complete(); err != nil {
+			return err
+		}
 	}
 
-	h := handler{
-		logger:    logger,
-		artifacts: artifacts,
+	return nil
+}
+
+var (
+	_ admission.CustomValidator = &backupBucketValidator{}
+	_ admission.CustomValidator = &backupEntryValidator{}
+	_ admission.CustomValidator = &bastionValidator{}
+	_ admission.CustomValidator = &containerRuntimeValidator{}
+	_ admission.CustomValidator = &controlPlaneValidator{}
+	_ admission.CustomValidator = &dnsRecordValidator{}
+	_ admission.CustomValidator = &etcdValidator{}
+	_ admission.CustomValidator = &extensionValidator{}
+	_ admission.CustomValidator = &infrastructureValidator{}
+	_ admission.CustomValidator = &networkValidator{}
+	_ admission.CustomValidator = &operatingSystemConfigValidator{}
+	_ admission.CustomValidator = &workerValidator{}
+
+	validators = map[client.Object]admission.CustomValidator{
+		&extensionsv1alpha1.BackupBucket{}:          &backupBucketValidator{},
+		&extensionsv1alpha1.BackupEntry{}:           &backupEntryValidator{},
+		&extensionsv1alpha1.Bastion{}:               &bastionValidator{},
+		&extensionsv1alpha1.ContainerRuntime{}:      &containerRuntimeValidator{},
+		&extensionsv1alpha1.ControlPlane{}:          &controlPlaneValidator{},
+		&extensionsv1alpha1.DNSRecord{}:             &dnsRecordValidator{},
+		&druidv1alpha1.Etcd{}:                       &etcdValidator{},
+		&extensionsv1alpha1.Extension{}:             &extensionValidator{},
+		&extensionsv1alpha1.Infrastructure{}:        &infrastructureValidator{},
+		&extensionsv1alpha1.Network{}:               &networkValidator{},
+		&extensionsv1alpha1.OperatingSystemConfig{}: &operatingSystemConfigValidator{},
+		&extensionsv1alpha1.Worker{}:                &workerValidator{},
 	}
+)
+
+type (
+	backupBucketValidator          struct{}
+	backupEntryValidator           struct{}
+	bastionValidator               struct{}
+	containerRuntimeValidator      struct{}
+	controlPlaneValidator          struct{}
+	dnsRecordValidator             struct{}
+	etcdValidator                  struct{}
+	extensionValidator             struct{}
+	infrastructureValidator        struct{}
+	networkValidator               struct{}
+	operatingSystemConfigValidator struct{}
+	workerValidator                struct{}
+)
+
+func (*backupBucketValidator) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	errors := validation.ValidateBackupBucket(obj.(*extensionsv1alpha1.BackupBucket))
+	return errors.ToAggregate()
```

Comment:
I would go for the following schema, which makes sure to correctly set `Invalid` as the response status reason:
```suggestion
	bb := obj.(*extensionsv1alpha1.BackupBucket)
	if errs := validation.ValidateBackupBucket(bb); len(errs) > 0 {
		return apierrors.NewInvalid(extensionsv1alpha1.Kind(extensionsv1alpha1.BackupBucketResource), bb.GetName(), errs)
	}
	return nil
```

See https://github.com/gardener/gardener/blob/c14f5d534e4e0a781bf6af0bb320da4e7a9c8e97/vendor/sigs.k8s.io/controller-runtime/pkg/webhook/admission/validator_custom.go#L101-L107